Consequences/Consequences/NSIStreamingProcessor.cs

```diff
@@ -1,51 +1,125 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Net.Http;
+using System.Reflection.Emit;
 using System.Text;
+using System.Text.Json;
+using System.Text.Json.Nodes;
 using System.Threading.Tasks;
 using USACE.HEC.Geography;
 
+
 namespace USACE.HEC.Consequences;
 public class NSIStreamingProcessor : IBBoxStreamingProcessor
 {
-  public void Process(BoundingBox boundingBox, Action<IConsequencesReceptor> ConsequenceReceptorProcess)
+  public async Task Process(BoundingBox boundingBox, Action<IConsequencesReceptor> ConsequenceReceptorProcess)
   {
-    Structure s = new Structure();
-    ConsequenceReceptorProcess(s);
+    await ProcessStream(boundingBox, ConsequenceReceptorProcess);
+    //await ProcessCollection(boundingBox, ConsequenceReceptorProcess);
   }
 
+  public async Task TestProcessStream(BoundingBox boundingBox, Action<IConsequencesReceptor> ConsequenceReceptorProcess)
+  {
+    await ProcessStream(boundingBox, ConsequenceReceptorProcess);
+  }
 
+  public async Task TestProcessCollection(BoundingBox boundingBox, Action<IConsequencesReceptor> ConsequenceReceptorProcess)
+  {
+    await ProcessCollection(boundingBox, ConsequenceReceptorProcess);
+  }
 
-  /*
-  static async Task Test()
+  private string ConstructURL(BoundingBox boundingBox, string directive)
   {
-    // Define the API endpoint
-    string apiEndpoint = "https://nsi.sec.usace.army.mil/nsiapi/structures?bbox=-81.58418,30.25165,-81.58161,30.26939,-81.55898,30.26939,-81.55281,30.24998,-81.58418,30.25165";
+    StringBuilder url = new StringBuilder();
+
+    url.Append("https://nsi.sec.usace.army.mil/nsiapi/structures?bbox=");
+    url.Append(boundingBox.NSIFormat());
+
+    // directive to specify collection or stream
+    url.Append(directive);
+
+    return url.ToString();
+  }
+
+  private async Task ProcessCollection(BoundingBox boundingBox, Action<IConsequencesReceptor> ConsequenceReceptorProcess)
+  {
+    // endpoint URL
+    string apiUrl = ConstructURL(boundingBox, "&fmt=fc");
 
     // Create an instance of HttpClient
     using (HttpClient client = new HttpClient())
```

Comment:
There may be a way to clean up this syntax. Just remove the curly braces. dotnet can infer the scope.

Reply:
Also, it's often cleaner to only specify the type once on a line if necessary. Shorthand for a constructor is just new(). Try that out